parseDataset.js

```diff
@@ -107,14 +108,20 @@ lines.slice(TITLE_LINE + 1).forEach((line) => {
     return;
   }
   if (isNewQuantity) {
+    // Default values
+    lineObject.quantity_unit_id = lineObject.quantity_unit_id || 'g';
+    lineObject.quantity_default_number_of_units = lineObject.quantity_default_number_of_units || 1;
```

Comment:
To fix the TS complaining that some `quantity_default_number_of_units` are not string


```suggestion
    lineObject.quantity_default_number_of_units = lineObject.quantity_default_number_of_units;
```
```suggestion
    lineObject.quantity_default_number_of_units = lineObject.quantity_default_number_of_units || "1";
```

But not sure why you are looking to  defaultized the `quantity_default_number_of_units` if the quantity is a weight it make sens to have this value undefined.


--- 

On this topic, a solution could be to add 
```js
const parseNumber = (n) => {
  if (n === undefined || Number.isNaN(n)) {
    return undefined;
  }
  return parseInt(n);
};
```

And then do a parsing such that numbers are saved as `number` and not strings. But that implies some other modification in the code

```suggestion
    // Parse numbers when provided
    lineObject.quantity_default_weight = parseNumber(
      lineObject.quantity_default_weight
    );
    lineObject.quantity_default_weight_per_unit = parseNumber(
      lineObject.quantity_default_weight_per_unit
    );
    lineObject.quantity_default_number_of_units = parseNumber(
      lineObject.quantity_default_number_of_units
    );
    lineObject.quantity_step = parseNumber(lineObject.quantity_step);
```

Reply:
You don't need it, if their is no default value (weight or unit) the `getDefaultQuantityValue` will return `1`

Reply:
By the way, I included those modification in #25